pyperformance/data-files/benchmarks/bm_docutils/run_benchmark.py

```diff
@@ -0,0 +1,60 @@
+"""
+Convert Docutils' documentation from reStructuredText to <format>.
+"""
+
+from pathlib import Path
+import shutil
+
+import docutils
+from docutils import core
+import pyperf
+
+try:
+    from docutils.utils.math.math2html import Trace
+except ImportError:
+    pass
+else:
+    Trace.show = lambda message, channel: ...  # don't print to console
+
+
+def build_html(doc_root, out_root):
+    for file in doc_root.rglob("*.txt"):
+        try:
+            dest = out_root / file.relative_to(doc_root).with_suffix(".html")
+            dest.parent.mkdir(parents=True, exist_ok=True)
+            core.publish_file(source_path=str(file),
```

Comment:
Is it possible to get `docutils` to use a string or `io.BytesIO` as input and output and read the input files into memory in advance so they aren't part of the measured time?  That would remove the effect of I/O (which can vary considerably) and only measure the CPU-bound parts of the benchmark.